### ISSUE TITLE
Style description parsing error/issue in darkroom view

### ISSUE THREAD

**Issue by jade-nl** [darktable-org/darktable#4196]
**Describe the bug**

When selecting a style that contains an ampersand (\&amp;) in its description field the information shown will be empty and the following message will appear in the terminal:

> (darktable:18847): Gtk-WARNING **: 11:31:14.179: Failed to set text '<b>Agfa (Black&White): Agfapan 400</b>
> channel mixer (on)' from markup due to error parsing markup: Error on line 1: Entity did not end with a semicolon; most likely you used an ampersand character without intending to start an entity ? escape ampersand as &amp;

There is, however, no lone & in the description field. It is coded the way it should be:

```
<info>
    <name>Black&amp;White|Agfa|Agfapan 400</name>
    <description>Agfa (Black&amp;White): Agfapan 400</description>
</info>
```

This only happens in darkroom view and only with the description field. No problems whatsoever in lighttable view and with the \&amp; in the name field.

**To Reproduce**

Create or change an existing style and use *\&amp;* in the description field.

0. Start darktable from terminal
1. In lighttable view
2. Open the styles module
3. Hover over the "problem" style
4. All information is shown, no messages in the terminal
5. Load an image in darkroom
6. Toggle the styles quick access
7. Hover over the "problem" style
8. No info is shown at all, multiple error messages in the terminal

**Expected behavior**
The information to be shown as is done when in lighttable view.

**Platform (please complete the following information):**
 - Darktable Version: Both 3.0.0 stable and 3.1.0+433
 - OS: Linux, Debian Buster (10.2)

**Additional context**

- I manually changed a style using vi and also tried creating one using darktable: Same result.
- This is not related to the hierarchical notation. Checked this.
- A clean environment (fully remove data.db, darktablerc etc)  doesn't help.
- Initially noticed on a personalised compilation (both stable and dev)
- Also present on a recompiled version using the build.sh script (using: ./build.sh --prefix /home/jade/dt.tmp)

**Comment by johnny-bit** [comment]
It seems that problem with reading description is not limited to `&amp`. I've managed to reproduce it using `&lt;`, however `&gt;` & `&quot;` do not produce the error mentioned.

**Comment by jade-nl** [comment]
I  did not check those......

There are only 5 characters that need to be escaped when using xml. There is still one missing from the above posts: \&apos; This one does not produce the issue.

Here are all 5 and the results when parsing them:

**_\&amp;_ error**
\&apos;  OK
\&gt; OK
**_\&lt;_ error**
\&quot;  OK

**Comment by TurboGit** [comment]
This is now fixed in master and 3.0.1. Thanks for reporting.
